tests/support_server/setup.pm

```diff
@@ -253,11 +253,15 @@ sub setup_xvnc_server {
 sub setup_xdmcp_server {
     return if $xdmcp_server_set;
 
+    if (check_var('REMOTE_DESKTOP_TYPE', 'xdmcp_xdm')) {
+        script_run "sed -i -e 's|^DISPLAYMANAGER=.*|DISPLAYMANAGER=\"xdm\"|' /etc/sysconfig/displaymanager";
+        script_run "sed -i -e 's|^DEFAULT_WM=.*|DEFAULT_WM=\"icewm\"|' /etc/sysconfig/windowmanager";
+    }
     script_run 'yast2 firewall services add zone=EXT service=service:xdmcp';
     script_run "sed -i -e 's|^DISPLAYMANAGER_REMOTE_ACCESS=.*|DISPLAYMANAGER_REMOTE_ACCESS=\"yes\"|' /etc/sysconfig/displaymanager";
     script_run "sed -i -e 's|^\\[xdmcp\\]|\\[xdmcp\\]\\nMaxSessions=2|' /etc/gdm/custom.conf";
     script_run 'systemctl restart display-manager.service';
-    assert_screen 'displaymanager';
+    assert_screen [qw(displaymanager displaymanager-xdm)];
```

Comment:
yes, I got this part. But I suggest you just check for one tag and give the same tag to all needles unless you are doing something specific based on the needle match.

If you would do e.g.

```perl
assert_screen [qw(foo bar)];
send_key 'baz' if match_has_tag 'foo';
```

then multiple tags would make sense.